drivers/video/ov5640.c

```diff
@@ -1244,81 +1246,23 @@ static int ov5640_init_controls(const struct device *dev)
 {
 	int ret;
 	struct ov5640_data *drv_data = dev->data;
-	struct ov5640_ctrls *ctrls = &drv_data->ctrls;
-
-	ret = video_init_ctrl(&ctrls->auto_gain, dev, VIDEO_CID_AUTOGAIN,
-			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 1});
-	if (ret) {
-		return ret;
-	}
-
-	ret = video_init_ctrl(
-		&ctrls->gain, dev, VIDEO_CID_ANALOGUE_GAIN,
-		(struct video_ctrl_range){.min = 0, .max = 1023, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
-
-	video_auto_cluster_ctrl(&ctrls->auto_gain, 2, true);
-
-	ret = video_init_ctrl(
-		&ctrls->brightness, dev, VIDEO_CID_BRIGHTNESS,
-		(struct video_ctrl_range){.min = -15, .max = 15, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
-
-	ret = video_init_ctrl(&ctrls->contrast, dev, VIDEO_CID_CONTRAST,
-			      (struct video_ctrl_range){.min = 0, .max = 255, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
-
-	ret = video_init_ctrl(&ctrls->hue, dev, VIDEO_CID_HUE,
-			      (struct video_ctrl_range){.min = 0, .max = 359, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
-
-	ret = video_init_ctrl(
-		&ctrls->saturation, dev, VIDEO_CID_SATURATION,
-		(struct video_ctrl_range){.min = 0, .max = 255, .step = 1, .def = 64});
-	if (ret) {
-		return ret;
-	}
-
-	ret = video_init_ctrl(&ctrls->hflip, dev, VIDEO_CID_HFLIP,
-			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
 
-	ret = video_init_ctrl(&ctrls->vflip, dev, VIDEO_CID_VFLIP,
-			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
-	if (ret) {
-		return ret;
-	}
+	struct video_ctrl *const ctrls[] = {
+#define FIELD_ADDR(name) &drv_data->ctrls.name,
+		OV5640_CTRLS(FIELD_ADDR)
+#undef FIELD_ADDR
+	};
 
-	ret = video_init_menu_ctrl(&ctrls->light_freq, dev, VIDEO_CID_POWER_LINE_FREQUENCY,
-				   VIDEO_CID_POWER_LINE_FREQUENCY_50HZ, NULL);
-	if (ret) {
-		return ret;
+	for (uint8_t i = 0; i < ARRAY_SIZE(ctrls); i++) {
+		ret = video_init_ctrl(ctrls[i], dev);
+		if (ret) {
+			return ret;
+		}
 	}
```

Comment:
It seems possible to do `ctrls = (struct video_ctrl *)&data->ctrls` and then loop like this:

```c
	for (uint8_t i = 0; i < sizeof(data->ctrls) / sizeof(*ctrls); i++) {
		ret = video_init_ctrl(ctrls[i], dev);
		if (ret) {
			return ret;
		}
	}
```